coding/url_helpers.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <functional>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace coding::url
```

Comment:
1. I'm almost ready to break the convention here and name this namespace plainly "url".
2. Could not move this to base/ (namespace base::url) which looks more suitable because there is a dependency on coding/hex.hpp. Does anybody have suggestions here?

Reply:
Мне кажется сейчас все хорошо. На мой взгляд даже лучше чем base, потому что построение url  это и есть кодирование. Разве нет?

Reply:
Плюсую по первому пункту. По второму идей нет.

Reply:
В каком-то смысле кодирование, но кажется, что пользоваться будет удобнее без дописывания coding::, а неоднозначности не возникнет (coding::url хорош тем, что сразу понятно, в каком проекте искать).

@milchakov @maksimandrianov так что, оставляем coding::url или делаем url:: ?

Reply:
мне кажется, что плюс 

> coding::url хорош тем, что сразу понятно, в каком проекте искать


больше, чем меньше писать:) но если вам так больше нравится, то ок, я не против